experimental/oauther/oauth2.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package oauther
+
+import (
+	"encoding/json"
+
+	"github.com/gorilla/mux"
+	"github.com/mattermost/mattermost-plugin-api/experimental/bot/logger"
+	"github.com/mattermost/mattermost-server/v5/plugin"
+	"golang.org/x/oauth2"
+)
+
+const (
+	oAuth2StateTimeToLive = 300 // seconds
```

Comment:
Should this be configurable?

Reply:
This is partly a magic number. 5 minutes feel more than enough to go through an OAuth process, and short enough to avoid being used in a wrong manner. I do not see many cases where this number must be larger or smaller, so adding some configuration for this would just mean more complex code and initialization without real usage.